### ISSUE TITLE
Old object sample description for types with length field

### ISSUE THREAD

**Issue by galderz** [oracle/graal#7569]
This is a follow-up enhancement to #5145.

Enhance old object sample event description to support types that have a `length` field. To achieve this, we would need to collect the field offset for each class during the image build and store it in the image heap. This information could then be used to provide the information on the types that have such a `length` field and provide equivalent description to that provided in HotSpot.

**Comment by kassifar** [comment]
Hello @galderz,

Thank you for your suggestion, we will take a look at it shortly.

**Comment by kassifar** [comment]
Hello @fniephaus, should I create an internal GR ticket for this ? Thank you.

**Comment by fniephaus** [comment]
@kassifar I'm not sure we need to. @galderz do you intend to work on this? Maybe to simplify, add the enhancement to #5145?

**Comment by galderz** [comment]
@fniephaus Yeah I plan to work on this, but I'd rather leave it as a follow up task because #5145 is big enough as it stands and I've had that PR open for several months now.

**Comment by galderz** [comment]
Also, this follow-up task is merely about the cosmetics of the description for a very small subset of types, so its priority is low.
